spec/controllers/admin_censor_rule_controller_spec.rb

```diff
@@ -280,18 +290,18 @@ def create_censor_rule
           expect(flash[:notice]).to eq(msg)
         end
 
-        it 'purges the cache for the info request' do
-          info_request = FactoryGirl.create(:info_request)
-          censor_rules = double
-          allow(info_request).to receive(:censor_rules) { censor_rules }
-          allow(InfoRequest).to receive(:find) { info_request }
-          censor_rule = FactoryGirl.build(:info_request_censor_rule, :info_request => info_request)
-          allow(censor_rules).to receive(:build) { censor_rule }
+        it 'calls expire_requests on the new censor_rule' do
+          allow(InfoRequest).to receive(:find) { @info_request }
```

Comment:
> My personal preference would be to actually create all these test objects and mock less things, is that possible? 

***This has taken me a while to work out and check to see what's going on - please correct me if I have any of this wrong***

Unfortunately because this a bit arm's length, the mocking is pretty awkward...

We need to stub a method on an object (the censor rule) which is created during the test (by the call to post) so ideally we need to send in a spy. A further complication is that the censor rule is built via a call on the info_request but it's not using *our* info_request object, it's calling `InfoRequest.find` with the id which returns the same row of the database but it turns out each call to `find` is treated as a different object which brings us back round to stubbing `InfoRequest.find` \*sigh\*.

A different way of doing this could be to use the [deprecated `allow_any_instance_of` syntax](https://github.com/rspec/rspec-mocks#settings-mocks-or-stubs-on-any-instance-of-a-class) to avoid stubbing the call to `InfoRequest.find` but that still leaves us with something fairly messy:

```rb
censor_rule_spy = FactoryGirl.build(:info_request_censor_rule,
                                    :info_request => info_request)
mock_rules = double()
allow(mock_rules).to receive(:build).and_return(censor_rule_spy)
allow_any_instance_of(InfoRequest).to receive(:censor_rules).
  and_return(mock_rules)

allow(censor_rule_spy).to receive(:expire_requests)

post :create, :censor_rule => censor_rule_params,
              :request_id => info_request.id

expect(censor_rule_spy).to have_received(:expire_requests)
```

(**note:** I've renamed `censor_rule` as `censor_rule_spy` in the code in an attempt to make the intention a bit clearer.)